github-adapter/src/main/java/be/xplore/githubmetrics/githubadapter/config/auth/GithubUnauthorizedInterceptor.java

```diff
@@ -0,0 +1,33 @@
+package be.xplore.githubmetrics.githubadapter.config.auth;
+
+import be.xplore.githubmetrics.githubadapter.exceptions.GithubRequestWasUnauthenticatedException;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+
+@Component
+public class GithubUnauthorizedInterceptor implements ClientHttpRequestInterceptor {
```

Comment:
This code probably belongs in a ResponseErrorHandler

Reply:
I changed it to use an implementation of ResponseErrorHandler, please give it a look if its okay like so.